docs/tutorials/dev_add_test_group.md

```diff
@@ -0,0 +1,2218 @@
+        btr_dt_per_km = config.getfloat('baroclinic_channel', 'btr_dt_per_km')
+        btr_dt = btr_dt_per_km * self.resolution
+        options['config_btr_dt'] = \
+            time.strftime('%H:%M:%S', time.gmtime(btr_dt))
+
+        self.dt = dt
+        self.btr_dt = btr_dt
+
+        self.add_model_config_options(options=options)
+```
+
+The default values for the polaris config options are again found in
+`baroclinic_channel.cfg`:
+
+```cfg
+# config options for baroclinic channel testcases
+[baroclinic_channel]
+
+# time step per resolution (s/km), since dt is proportional to resolution
+dt_per_km = 30
+
+# barotropic time step per resolution (s/km), since btr_dt is proportional to
+# resolution
+btr_dt_per_km = 1.5
+```
+
+Unlike in `compute_cell_count()`, we don't do anything differently in
+`dynamic_model_config()` here whether it's run at setup or at runtime.  The
+reason we run it twice is to update the model config options in case the user
+modified `dt_per_km` or `btr_dt_per_km` in the config file in the work 
+directory before running the step.
+
+## Adding a visualization step
+
+We'll add one more step to make some plots after the forward run has finished.
+Here is the contents of `viz.py`:
+
+```python
+import cmocean  # noqa: F401
+import numpy as np
+import xarray as xr
+
+from polaris import Step
+from polaris.viz import plot_horiz_field
+
+
+class Viz(Step):
+    """
+    A step for plotting the results of a series of RPE runs in the baroclinic
+    channel test group
+    """
+    def __init__(self, test_case):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : polaris.TestCase
+            The test case this step belongs to
+        """
+        super().__init__(test_case=test_case, name='viz')
+        self.add_input_file(
+            filename='initial_state.nc',
+            target='../initial_state/initial_state.nc')
+        self.add_input_file(
+            filename='output.nc',
+            target='../forward/output.nc')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        ds_mesh = xr.load_dataset('initial_state.nc')
+        ds = xr.load_dataset('output.nc')
+        t_index = ds.sizes['Time'] - 1
+        plot_horiz_field(ds, ds_mesh, 'temperature',
+                         'final_temperature.png', t_index=t_index)
+        max_velocity = np.max(np.abs(ds.normalVelocity.values))
+        plot_horiz_field(ds, ds_mesh, 'normalVelocity',
+                         'final_normalVelocity.png',
+                         t_index=t_index,
+                         vmin=-max_velocity, vmax=max_velocity,
+                         cmap='cmo.balance', show_patch_edges=True)
+```
+
+It makes images of the final temperature and normal velocity from a forward
+step.  Since all the pieces of this step have been covered in the other 2 
+steps, we won't describe this step in any more detail.
+
+### Adding the steps to the test case
```

Comment:
In terms of using the tutorial to make a new test case on the fly, I prefer to add each step to the test case as I've completed it and test that it works before moving onto the next one. So this section would be replicated earlier for just the `initial_state` step, then adding `forward`